### PR TITLE
Add an empty reboot weak stub to archs that do not have it, update `shell_module` test to build more

### PR DESCRIPTION
**Pull request by ycsin**
- Add weak `sys_arch_reboot()` to archs that do not have it, so that `CONFIG_REBOOT` can be enabled freely in apps
- Updated the test in `shell_module` sample to build everything inside the `kernel_service/` on qemu_riscv64

### CODE REVIEW

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
> Add weak sys_arch_reboot() to archs that do not have it, so that CONFIG_REBOOT can be enabled freely in apps

What's the idea here? Adding a fake reboot hook will render applications depending on reboot functionality happy until they call it and nothing happens?

**Comment by ycsin** [comment]
> > Add weak sys_arch_reboot() to archs that do not have it, so that CONFIG_REBOOT can be enabled freely in apps
> 
> What's the idea here? Adding a fake reboot hook will render applications depending on reboot functionality happy until they call it and nothing happens?

That's right, application / board / soc can implement the function and override the stub if necessary, this isn't something new, see #31901, #77905 

The `sys_reboot()` would make sure that something happens:

https://github.com/zephyrproject-rtos/zephyr/blob/7773fe8dd60e9f86760c994090304a0577a1ad2f/lib/os/reboot.c#L39-L45

File: arch/nios2/core/CMakeLists.txt [hunk 1]
```diff
@@ -17,3 +17,4 @@ zephyr_library_sources(
 
 zephyr_library_sources_ifdef(CONFIG_IRQ_OFFLOAD irq_offload.c)
 zephyr_library_sources_ifdef(CONFIG_TIMING_FUNCTIONS timing.c)
+zephyr_library_sources_ifdef(CONFIG_REBOOT reboot.c)
```
Comment (jilaypandya) [inline arch/nios2/core/CMakeLists.txt:20]: arrange alphabetically

**Comment by bjarki-andreasen** [comment]
I don't quite agree with this. Comparing it to sys poweroff:
```
config POWEROFF
	bool "Power off functionality"
	depends on HAS_POWEROFF
	help
	  Enable support for system power off.
```
I would argue we are missing a `HAS_REBOOT`, which would prevent the IMO invalid build configuration of enabling reboot while the arch does not support it

**Review by gmarull — CHANGES_REQUESTED** [review]
See https://github.com/zephyrproject-rtos/zephyr/pull/60920 (you may want to retake it)

**Comment by ycsin** [comment]
Maybe next time

**Pull request closed without merging**